server/src/main/java/io/druid/server/emitter/HttpEmitterModule.java

```diff
@@ -101,10 +97,53 @@ public Emitter getEmitter(
             createAsyncHttpClient(
                 "HttpPostEmitter-AsyncHttpClient-%d",
                 "HttpPostEmitter-AsyncHttpClient-Timer-%d",
-                sslContext
+                getEffectiveSSLContext(sslConfig.get(), sslContext)
             )
         ),
         jsonMapper
     );
   }
+
+  public static SSLContext getEffectiveSSLContext(HttpEmitterSSLClientConfig sslConfig, SSLContext sslContext)
+  {
+    SSLContext effectiveSSLContext;
+    if (sslConfig.isUseDefaultJavaContext()) {
+      try {
+        effectiveSSLContext = SSLContext.getDefault();
+      }
+      catch (NoSuchAlgorithmException nsae) {
+        throw new RuntimeException(nsae);
+      }
+    } else if (sslConfig.getTrustStorePath() != null) {
+      log.info("Creating SSLContext for HttpEmitter client using config [%s]", sslConfig);
+      effectiveSSLContext = getSSLContextFromConfig(sslConfig);
+    } else {
+      effectiveSSLContext = sslContext;
+    }
+    return effectiveSSLContext;
+  }
+
+  public static SSLContext getSSLContextFromConfig(HttpEmitterSSLClientConfig config)
```

Comment:
looks pretty much same as SSLContextProvider.get(). Can we refactor the code a bit and created a static method in SSLContextProvider that could be used here ?

Reply:
I moved this block to a utility class, SSLContextProvider is in an extension